### PR TITLE
Update oc_sec_apply_cred callback

### PR DESCRIPTION
**Pull request by Danielius1922**

### CODE REVIEW

File: security/oc_cred.c [hunk 1]
```diff
@@ -1249,9 +1307,17 @@ oc_sec_decode_cred(oc_rep_t *rep, oc_sec_cred_t **owner, bool from_storage,
...
+                oc_sec_on_apply_cred_data_t cred_data = {
+                  cr,
+                  add_cred_data.replaced_cred,
+                  add_cred_data.created,
+                };
+                oc_apply_cred_cb(cred_data, on_apply_cred_data);
               }
             }
+            if (add_cred_data.replaced_cred) {
+              oc_sec_free_cred(add_cred_data.replaced_cred);
```
Comment (jkralik) [inline security/oc_cred.c:1319]: maybe we need to free it at line 1294:
```c
  if (!oscore_ctx) {
   return false;
  }
```

Comment (Danielius1922) [reply security/oc_cred.c:1319]: Good catch, I recall adding it there, but somehow it disappeared. Fixed.

**Review by ondrejtomcik — APPROVED** [review]

File: security/oc_cred.c [hunk 2]
```diff
@@ -128,28 +128,58 @@ is_existing_cred(int credid, bool roles_resource, oc_tls_peer_t *client,
...
+    ((authority_len == oc_string_len(cred->role.authority)) &&
+     (memcmp(authority, oc_string(cred->role.authority), authority_len) == 0));
+}
+
+static bool
+oc_sec_role_cred_match_tag(const oc_sec_cred_t *cred, const char *tag,
+                           size_t tag_len)
+{
+  return // skip check
+    (tag_len == 0) ||
```
Comment (jkralik) [inline security/oc_cred.c:158]: tag == `NULL`

**Comment by ondrejtomcik** [comment]
@WAvdBeek order n2

**Review by jkralik — APPROVED** [review]

**Comment by Danielius1922** [comment]
@SiMet could you please run CTT test with this branch before Monday? (So we have this and #233 in master on Monday). Thanks.

**Comment by SiMet** [comment]
> @SiMet could you please run CTT test with this branch before Monday? (So we have this and #233 in master on Monday). Thanks.

I've just trigger the job

**Comment by SiMet** [comment]
Hi all,

Both server and client simulator failed in test 1.7.4.5.
Please find logs: [fails.zip](https://github.com/iotivity/iotivity-lite/files/8459760/fails.zip)

I'll analyse and provide more info tomorrow.

**Comment by ondrejtomcik** [comment]
@SiMet and is current master passing?

**Comment by SiMet** [comment]
> @SiMet and is current master passing?

Yes, current master is passing.

**Comment by Danielius1922** [comment]
> Hi all,
> 
> Both server and client simulator failed in test 1.7.4.5. Please find logs: [fails.zip](https://github.com/iotivity/iotivity-lite/files/8459760/fails.zip)
> 
> I'll analyse and provide more info tomorrow.

Hello, would it be possible to send me the logs of the successful master run? It seems that there some issue with the /oic/sec/roles resource. I'll go over the code, but if I compare the logs of the failed and successful runs maybe I'll get some hints about what's precisely the issue.

**Comment by SiMet** [comment]
> > Hi all,
> > Both server and client simulator failed in test 1.7.4.5. Please find logs: [fails.zip](https://github.com/iotivity/iotivity-lite/files/8459760/fails.zip)
> > I'll analyse and provide more info tomorrow.
> 
> Hello, would it be possible to send me the logs of the successful master run? It seems that there some issue with the /oic/sec/roles resource. I'll go over the code, but if I compare the logs of the failed and successful runs maybe I'll get some hints about what's precisely the issue.
Hi,

Please find passing logs from master: [CTT_Logs_Archive.zip](https://github.com/iotivity/iotivity-lite/files/8462390/CTT_Logs_Archive.zip)

I found that in failing scenario CTT does not receive response for the request and due to this testcase is aborted.

**Comment by Danielius1922** [comment]
@SiMet I found something that I'd like to verify. I have the CTT tool, but I've only ever run `OCF Cloud Devices` / `Origin Cloud` and `Target Cloud` tests. I don't see other test suites. Is there a way to run the failing 1.7.4.5 test case?

**Comment by SiMet** [comment]
Probably you need to select correct PICS file.

You could run client_certification_tests app from: https://github.com/iotivity/iotivity-lite/tree/master/apps 

and provide PICS:
```{
    "device": "Client",
    "company": "Company_name",
    "platformVersion": "1.0",
    "role": "Client",
    "icv": "ocf.2.2.5",
    "dmv": "ocf.res.1.3.0, ocf.sh.1.3.0",
    "supportedVerticalProfile": [
        "Smart Home"
    ],
    "supportedDeviceTypes": [
        "oic.d.light"
    ],
    "resources": [
        "oic.r.sdi",
        "oic.wk.res",
        "oic.r.csr",
        "oic.r.ael",
        "oic.r.sp",
        "oic.r.roles",
        "oic.wk.mnt",
        "oic.wk.con",
        "oic.r.softwareupdate"
    ],
    "optionalFeatures": [
        "batchInterface",
        "ipv6Scope3",
        "ipv6Scope5",
        "supportsWellknownCoreDiscovery"
    ],
    "OTM": [
        "oic.sec.doxm.jw",
        "oic.sec.doxm.rdp",
        "oic.sec.doxm.mfgcert"
    ],
    "contentFormatVersion": [
        "1.0.0"
    ],
    "acceptVersion": [
        "1.0.0"
    ],
    "supportedSecurityProfiles": [
        "sp-baseline-v0",
        "sp-black-v0",
        "sp-blue-v0",
        "sp-purple-v0"
    ],
    "cplAtIANAPen": "1.3.6.1.4.1.71",
    "cplAtModel": "Discovery",
    "cplAtVersion": "1.0",
    "multiValueQuerySupport": false,
    "sct": 457,
    "supportsCoAPoverTCP": true,
    "canTurnOnSoftApManually": false,
    "observableOICRES": true,
    "supportsCancellingObserve": true,
    "persistentDeviceuuid": false,
    "onboardingRoles": [
        
    ],
    "onboardingOTM": [
        
    ],
    "onboardingSpec": [
        
    ],
    "PasswordSoftAP": null,
    "RDPublishCapability": false,
    "SSIDSoftAP": null,
    "atomicMeasurements": [
        
    ],
    "authorizationServerUrl": null,
    "cis": null,
    "cloudAccountPassword": null,
    "cloudAccountUrl": null,
    "cloudAccountUser": null,
    "cloudClientCertificate": null,
    "cloudClientPrivateKey": null,
    "cloudServerTrustAnchorCertificate": null,
    "deviceApiUrl": null,
    "jurisdictionSwitch": false,
    "maxNotifyPeriod": [
        
    ],
    "minNotifyPeriod": [
        
    ],
    "redirectUri": null,
    "sid": null,
    "threshold": [
        
    ],
    "threshold_recommended_value": 0,
    "validClientId": null,
    "validClientSecret": null,
    "supportsMultipleDTLS": true
}
```

**Comment by SiMet** [comment]
If you use server_certification_app runned on the same network as CTT then you should see the device on the list in IUT discovery.

image

then please use PICS:
```
{
    "OTM": [
        "oic.sec.doxm.jw",
        "oic.sec.doxm.rdp",
        "oic.sec.doxm.mfgcert"
    ],
    "PasswordSoftAP": "",
    "RDPublishCapability": true,
    "SSIDSoftAP": "",
    "acceptVersion": [
        "1.0.0"
    ],
    "atomicMeasurements": [
        
    ],
    "canTurnOnSoftApManually": true,
    "cloudRegisterAutoRetry": false,
    "company": "Company_name",
    "contentFormatVersion": [
        "1.0.0"
    ],
    "cplAtIANAPen": "1.3.6.1.4.1.71",
    "cplAtModel": "Discovery",
    "cplAtVersion": "1.0",
    "device": "Cloud_proxy",
    "dmv": "ocf.res.1.3.0, ocf.sh.1.3.0",
    "eSIMTimeout": 0,
    "icv": "ocf.2.2.5",
    "jurisdictionSwitch": false,
    "maxNotifyPeriod": [
        0,
        20000
    ],
    "minNotifyPeriod": [
        0,
        10000
    ],
    "multiValueQuerySupport": true,
    "observableOICRES": false,
    "onboardingOTM": [
        
    ],
    "onboardingRoles": [
        
    ],
    "onboardingSpec": [
        
    ],
    "optionalFeatures": [
        "batchInterface",
        "supportsWellknownCoreDiscovery"
    ],
    "persistentDeviceuuid": false,
    "resources": [
        "oic.r.sdi",
        "oic.r.sp",
        "oic.wk.res",
        "oic.r.csr",
        "oic.r.roles",
        "oic.r.ael",
        "oic.r.coapcloudconf",
        "oic.r.d2dserverlist"
    ],
    "role": "Server",
    "sct": 457,
    "supportedDeviceTypes": [
        "oic.d.cloudproxy"
    ],
    "supportedSecurityProfiles": [
        "sp-baseline-v0",
        "sp-black-v0",
        "sp-blue-v0",
        "sp-purple-v0"
    ],
    "supportedVerticalProfile": [
        "Smart Home"
    ],
    "supportsCancellingObserve": false,
    "supportsCoAPoverTCP": true,
    "supportsMultipleDTLS": true,
    "supportseSIMEasySetup": false,
    "threshold": [
        0,
        5
    ],
    "threshold_recommended_value": 2
}
```

in case you'd like to test client then after start client_certification_app and CTT -> IUT Selection you need to send Discovery from the iotivity-lite application

image

**Comment by Danielius1922** [comment]
Thank you @SiMet, I was able to make it work (thanks also to @jkralik). (I shall not mention that most of my issues was caused by having a VPN automatically enabled on start-up and not thinking of it :X ). Can you please run the full CTT test suite again?

**Comment by SiMet** [comment]
All testcases passed :)

**Comment by jkralik** [comment]
@SiMet 👍 
@WAvdBeek Pls could you check PR ? Thx

**Comment by ondrejtomcik** [comment]
@SiMet can you inform us when master passes CTT tests? Wouter merged tinyCBOR before this one. Thanks!

**Comment by SiMet** [comment]
> @SiMet can you inform us when master passes CTT tests? Wouter merged tinyCBOR before this one. Thanks!

I'll have results tomorrow.

**Comment by SiMet** [comment]
Hi,

@ondrejtomcik, @WAvdBeek 
We have build crash today:
```
23:06:01 + make IPV4=1 CLOUD=1 SECURE=0 OSCORE=0 DEBUG=0 cloud_tests
23:06:01 OBJ_PYTHON is obj/python/oc_python.o
23:06:01 SRC_PYTHON is ../../python/oc_python.c
23:06:01 gcc -c -o obj/oc_mem_trace.o ../../util/oc_mem_trace.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
23:06:01 gcc -c -o obj/oc_timer.o ../../util/oc_timer.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
23:06:01 gcc -c -o obj/oc_mmem.o ../../util/oc_mmem.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
23:06:01 gcc -c -o obj/oc_list.o ../../util/oc_list.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
23:06:01 gcc -c -o obj/oc_etimer.o ../../util/oc_etimer.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
23:06:01 gcc -c -o obj/oc_memb.o ../../util/oc_memb.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
23:06:01 gcc -c -o obj/oc_process.o ../../util/oc_process.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
23:06:01 gcc -c -o obj/cborencoder.o ../../deps/tinycbor/src/cborencoder.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
23:06:01 gcc -c -o obj/cborencoder_close_container_checked.o ../../deps/tinycbor/src/cborencoder_close_container_checked.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
23:06:01 gcc -c -o obj/cborencoder_float.o ../../deps/tinycbor/src/cborencoder_float.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
23:06:01 gcc: error: ../../deps/tinycbor/src/cborencoder_float.c: No such file or directory
23:06:01 Makefile:329: recipe for target 'obj/cborencoder_float.o' failed
23:06:01 make: *** [obj/cborencoder_float.o] Error 1
```

**Comment by ondrejtomcik** [comment]
@WAvdBeek can you please revert tinycbor and merge this one? Tinycbor is not blocking one, this one and acl's are.

**Comment by SiMet** [comment]
> Hi,
> 
> @ondrejtomcik, @WAvdBeek We have build crash today:
> 
> ```
> 23:06:01 + make IPV4=1 CLOUD=1 SECURE=0 OSCORE=0 DEBUG=0 cloud_tests
> 23:06:01 OBJ_PYTHON is obj/python/oc_python.o
> 23:06:01 SRC_PYTHON is ../../python/oc_python.c
> 23:06:01 gcc -c -o obj/oc_mem_trace.o ../../util/oc_mem_trace.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
> 23:06:01 gcc -c -o obj/oc_timer.o ../../util/oc_timer.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
> 23:06:01 gcc -c -o obj/oc_mmem.o ../../util/oc_mmem.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
> 23:06:01 gcc -c -o obj/oc_list.o ../../util/oc_list.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
> 23:06:01 gcc -c -o obj/oc_etimer.o ../../util/oc_etimer.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
> 23:06:01 gcc -c -o obj/oc_memb.o ../../util/oc_memb.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
> 23:06:01 gcc -c -o obj/oc_process.o ../../util/oc_process.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
> 23:06:01 gcc -c -o obj/cborencoder.o ../../deps/tinycbor/src/cborencoder.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
> 23:06:01 gcc -c -o obj/cborencoder_close_container_checked.o ../../deps/tinycbor/src/cborencoder_close_container_checked.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
> 23:06:01 gcc -c -o obj/cborencoder_float.o ../../deps/tinycbor/src/cborencoder_float.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
> 23:06:01 gcc: error: ../../deps/tinycbor/src/cborencoder_float.c: No such file or directory
> 23:06:01 Makefile:329: recipe for target 'obj/cborencoder_float.o' failed
> 23:06:01 make: *** [obj/cborencoder_float.o] Error 1
> ```

It's probably issue in our build machine that we do not updated submodule.
We investigate this and try prepare fix for that.

We found that in make there is step to update submodules but for mbedtls target.
This failed in case build with flag SECURE=0 when mbedtls is not build.

**Comment by SiMet** [comment]
> > Hi,
> > @ondrejtomcik, @WAvdBeek We have build crash today:
> > ```
> > 23:06:01 + make IPV4=1 CLOUD=1 SECURE=0 OSCORE=0 DEBUG=0 cloud_tests
> > 23:06:01 OBJ_PYTHON is obj/python/oc_python.o
> > 23:06:01 SRC_PYTHON is ../../python/oc_python.c
> > 23:06:01 gcc -c -o obj/oc_mem_trace.o ../../util/oc_mem_trace.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
> > 23:06:01 gcc -c -o obj/oc_timer.o ../../util/oc_timer.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
> > 23:06:01 gcc -c -o obj/oc_mmem.o ../../util/oc_mmem.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
> > 23:06:01 gcc -c -o obj/oc_list.o ../../util/oc_list.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
> > 23:06:01 gcc -c -o obj/oc_etimer.o ../../util/oc_etimer.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
> > 23:06:01 gcc -c -o obj/oc_memb.o ../../util/oc_memb.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
> > 23:06:01 gcc -c -o obj/oc_process.o ../../util/oc_process.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
> > 23:06:01 gcc -c -o obj/cborencoder.o ../../deps/tinycbor/src/cborencoder.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
> > 23:06:01 gcc -c -o obj/cborencoder_close_container_checked.o ../../deps/tinycbor/src/cborencoder_close_container_checked.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
> > 23:06:01 gcc -c -o obj/cborencoder_float.o ../../deps/tinycbor/src/cborencoder_float.c -fPIC -fno-asynchronous-unwind-tables -fno-omit-frame-pointer -ffreestanding -Os -fno-stack-protector -ffunction-sections -fdata-sections -fno-reorder-functions -fno-defer-pop -fno-strict-overflow -I./ -I../../include/ -I../../ -I../../api -std=gnu99 -Wall -Wextra -Werror -pedantic  -DOC_CLOUD -Wl,--gc-sections  -DOC_PKI -DOC_DYNAMIC_ALLOCATION -DOC_IDD_API -DOC_IPV4 -DOC_TCP
> > 23:06:01 gcc: error: ../../deps/tinycbor/src/cborencoder_float.c: No such file or directory
> > 23:06:01 Makefile:329: recipe for target 'obj/cborencoder_float.o' failed
> > 23:06:01 make: *** [obj/cborencoder_float.o] Error 1
> > ```
> 
> It's probably issue in our build machine that we did not update submodule. We investigate this and try prepare fix for that.
> 
> We found that in make there is step to update submodules but for mbedtls target. This failed in case build with flag SECURE=0 when mbedtls is not build.

I confirm. It's the issue of our test machine env. We preapred the fix. Master works correctly.

**Pull request merged**